src/decision/include/DecisionNode.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Created By: Reid Oliveira
+ * Created On: March 17, 2018
+ * Description:
+ */
+#ifndef DECISION_DECISION_H_H
+#define DECISION_DECISION_H_H
+
+#include "Subroutine.h"
+#include <geometry_msgs/Twist.h>
+#include <ros/ros.h>
+#include <unordered_map>
+
+// state msg
+#include <worldstate/state_msg.h>
+// subroutines
+#include "LineUpWithGate.h"
+#include "LocateGate.h"
+
+typedef int8_t state_t;
+
+class DecisionNode {
+  public:
+    DecisionNode(int argc, char** argv, std::string node_name);
+
+  private:
+    std::unordered_map<state_t, Subroutine*> subroutines_;
+    Subroutine* running_;
+    ros::Subscriber subscriber_;
+
+    void subscriberCallback(const worldstate::state_msg::ConstPtr& msg);
+    void setupSubroutineMap(int argc, char** argv);
```

Comment:
I'm not sure if we've settled on a standard for this, but it would be helpful if all functions had some sort of comment (javadoc, doxygen, etc.) to help make their functionality clear. Same goes for all function declarations (not just ones in this file).

Reply:
I don't think we've chosen one. I've made the decision to go with javadoc, because I find it a lot nicer. Done.